### PR TITLE
add Holonix upgrade guide

### PR DESCRIPTION
**Pull request by pdaoust**

### CODE REVIEW

File: src/pages/resources/upgrade/upgrade-new-holonix.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+title: Upgrade to the new Holonix
+---
+
+::: intro
+In the third quarter of 2024, we released a new [Holonix](/get-started/install-advanced/#holonix-the-holochain-app-development-environment), our development environment based on [Nix](https://nixos.org) that gives you all the dependencies to build hApps and contribute to Holochain core development.
```
Comment (jost-s) [inline src/pages/resources/upgrade/upgrade-new-holonix.md:6]: We're not using Nix as a basis for Holochain development any more. The new Holonix is just for hApp development.
```suggestion
In the third quarter of 2024, we released a new [Holonix](/get-started/install-advanced/#holonix-the-holochain-app-development-environment), our development environment based on [Nix](https://nixos.org) that gives you all the dependencies to build hApps.
```

File: src/pages/resources/upgrade/upgrade-new-holonix.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+## Upgrade an existing project
+
+The previous and the new Holonix distributions are both based on Nix's [flakes](https://wiki.nixos.org/wiki/Flakes) feature. Fortunately, this means it's easy to upgrade. First back up and remove your project's existing flake files:
+
+```bash
+mv flake.nix flake.nix.backup
+mv flake.lock flake.lock.backup
+```
+
+Then create new flake files:
```
Comment (jost-s) [inline src/pages/resources/upgrade/upgrade-new-holonix.md:20]: The flake.lock file is not generated until the flake is invoked for the first time.
```suggestion
Then create a new flake file:
```

File: src/pages/resources/upgrade/upgrade-new-holonix.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+...
+inputs = {
+-   holonix.url = "github:holochain/holonix?ref=main-0.3";
++   holonix.url = "github:holochain/holonix?ref=main-0.2";
+    ...
+};
+...
+```
+
+Next time you type `nix develop` to enter the shell, it'll update your `nix.lock` and download the requested packages for you.
```
Comment (jost-s) [inline src/pages/resources/upgrade/upgrade-new-holonix.md:56]:
```suggestion
Next time you type `nix develop` to enter the shell, it'll update your `flake.lock` and download the requested packages for you.
```

**Review by jost-s — APPROVED** [review]
Easy to follow and understand, I think.

The sections after "Upgrade an existing project" are similar to the README in the Holonix repro. I prefer to link to that README from here or add multiple links to anchors on that README instead. If you think there should be something added/clarified there, I'd be delighted about a PR there.

**Comment by pdaoust** [comment]
@jost-s 

> I prefer to link to that README from here or add multiple links to anchors on that README instead.

sounds like a good idea; I'll do that.

**Pull request merged**